tuned/exports/unix_socket_exporter.py

```diff
@@ -0,0 +1,228 @@
+			except OSError as e:
+				log.warning("Error while sending signal '%s' to socket '%s': %s" % (signal, p, e))
+
+	def register_signal_path(self, path):
+		self._socket_signal_paths.append(path)
+
+	def _construct_socket_object(self):
+		if self._socket_path:
+			if os.path.exists(self._socket_path):
+				os.unlink(self._socket_path)
+			s = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+			s.setblocking(False)
+			s.bind(self._socket_path)
+			os.chown(self._socket_path, self._ownership[0], self._ownership[1])
+			if self._permissions:
+				os.chmod(self._socket_path, self._permissions)
+			self._socket_object = s
+
+	def start(self):
+		if self.running():
+			return
+
+		self.stop()
+		self._construct_socket_object()
+
+	def stop(self):
+		if self._socket_object:
+			self._socket_object.close()
+
+	def _send_data(self, s, address, type, data):
+		"""
+		Sends data via socket s to address.
+		First byte of message specifies type of message (currently "r" for response, "e" for error message and "s" for signal).
+		Four more bytes are integer in network/big-endian with length of rest of message.
+		Rest of message is json data.
+		"""
+		log.debug("Sending socket data (type: %s, data: %s) to %s" % (type, data, address))
+		try:
+			j = json.dumps(data).encode("utf-8")
+			s.sendto(struct.pack("!sL", type.encode("utf-8"), len(j)), address)
+			s.sendto(j, address)
+		except Exception as e:
+			log.warning("Failed send data '%s' to '%s' type '%s': %s" % (data, address, type, e))
+
+	def period_check(self):
+		"""
+		Periodically checks socket object for new calls. This allows to function without special thread.
+		First four bytes is expected to be integer in network/big-endian with length of rest of message.
+		Rest of message expects json array, where first item is name of called method, rest is arguments of given method.
+		For response see method _send_data.
+		
+		Example implementation:
+		
+		def sss(d):
+			ssock_file = "/tmp/tuned_socket"
+			csock_file = "/tmp/tuned_socket_client"
+			if os.path.exists(csock_file):
+				os.remove(csock_file)
+			s=socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+			s.bind(csock_file)
+			j = json.dumps(d).encode("utf-8")
+			try:
+				s.sendto(struct.pack("!L", len(j)), ssock_file)
+				s.sendto(j, ssock_file)
+				d = s.recvfrom(5)
+				t, l = struct.unpack("!sL", d[0])
+				print("TYPE: %s" % t)
+				print("LEN: %s" % l)
+				data = json.loads(s.recvfrom(l)[0])
+				print("DATA: %s" % data)
+			except Exception as e:
+				print("E: %s" % e)
+			s.close()
+			
+		sss(("profiles",))
+		sss(("switch_profile", "balanced"))
+		"""
+		r, _, _ = select.select([self._socket_object], (), (), 0)
+		if r:
+			try:
+				l, address = self._socket_object.recvfrom(BYTES_LEN_OF_MSG)
+				l = struct.unpack("!L", l)[0]
+				log.debug("%s %s" % (address, l))
+			except Exception as e:
+				log.error("Failed to load length of message: %s" % e)
+				return
+			try:
+				data, address = self._socket_object.recvfrom(l)
```

Comment:
Just use a typical max datagram size: 4096. It will read less when the datagram is smaller.